.github/workflows/ci.yml

```diff
@@ -0,0 +1,134 @@
+## (5b) Pin to commits, not tags
+
+name: CI Build and Check
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+env:
+  IMAGE: rancher/kube-api-auth
+
+permissions:
+  contents: read
+  security-events: write # upload Trivy Sarif results
+
+jobs:
+  linter:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Get Sources
+      uses: actions/checkout@v3
+
+    - name: Set up Go
+      uses: actions/setup-go@v5
+      with:
+        go-version: '1.22'
+        cache: false
+
+    - name: Tidy and verify modules
+      run: |
+        echo Tidying up modules
+        go mod tidy
+        echo Verifying modules
+        go mod verify
```

Comment:
I'm also going through migrating the CI stuff. Would it be better to run `go mod verify` without running `go mod tidy` to detect whether the pushed code has a valid `go.mod` file?

Reply:
Nevermind, just tested it and it looks like go mod verify wouldn't spot that.

Reply:
Most of the commands in the flow are straight out of the `scripts/*` files.
The `go mod ...` commands originate from `scripts/validate`.